### PR TITLE
fix: #170 compile fail on aarch64

### LINKED ISSUE #170: Compile error on aarch64 in version 0.8.4
```
error[E0423]: expected function, found macro `transmute`
   --> /Users/purewhite/.cargo/registry/src/rsproxy.cn-0dccff568467c15b/ahash-0.8.4/src/operations.rs:128:15
    |
128 |         xor ^ transmute::<_, u128>(vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))))
    |               ^^^^^^^^^^^^^^^^^^^^ not a function
    |
help: use `!` to invoke the macro
    |
128 |         xor ^ transmute::<_, u128>!(vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))))
    |                                   +
help: consider importing one of these items instead
    |
1   + use core::mem::transmute;
    |
1   + use std::mem::transmute;
    |
1   + use zerocopy::macro_util::core_reexport::mem::transmute;
    |

error[E0423]: expected function, found macro `transmute`
   --> /Users/purewhite/.cargo/registry/src/rsproxy.cn-0dccff568467c15b/ahash-0.8.4/src/operations.rs:161:15
    |
161 |         xor ^ transmute::<_, u128>(vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))))
    |               ^^^^^^^^^^^^^^^^^^^^ not a function
    |
help: use `!` to invoke the macro
    |
161 |         xor ^ transmute::<_, u128>!(vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))))
    |                                   +
help: consider importing one of these items instead
    |
1   + use core::mem::transmute;
    |
1   + use std::mem::transmute;
    |
1   + use zerocopy::macro_util::core_reexport::mem::transmute;
    |

For more information about this error, try `rustc --explain E0423`.
error: could not compile `ahash` (lib) due to 2 previous errors
```

env:

```
rustc 1.75.0-nightly (42b1224e9 2023-10-15)
macOS 14.0 with Apple Silicon
```

### PR DESCRIPTION
**Pull request by PureWhiteWu**
fixes #170

### CODE REVIEW

File: src/operations.rs [hunk 1]
```diff
@@ -156,6 +157,7 @@ pub(crate) fn aesdec(value: u128, xor: u128) -> u128 {
     use core::arch::aarch64::*;
     #[cfg(target_arch = "arm")]
     use core::arch::arm::*;
+    use core::mem::transmute;
     unsafe {
         let value = transmute!(value);
         xor ^ transmute::<_, u128>(vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))))
```
Comment (joshlf) [inline src/operations.rs:163]: I'd recommend using `transmute!` so we can remove the `unsafe` here:

```suggestion
    let value = transmute!(value);
    let value: u128 = transmute!(vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))))
    xor ^ value
```

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/operations.rs [hunk 2]
```diff
@@ -123,6 +123,7 @@ pub(crate) fn aesenc(value: u128, xor: u128) -> u128 {
     use core::arch::aarch64::*;
     #[cfg(target_arch = "arm")]
     use core::arch::arm::*;
+    use core::mem::transmute;
     unsafe {
         let value = transmute!(value);
         xor ^ transmute::<_, u128>(vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))))
```
Comment (joshlf) [inline src/operations.rs:129]: I'd recommend using `transmute!` so we can remove the `unsafe` here:

```suggestion
    let value = transmute!(value);
    let value: u128 = transmute!(vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))));
    xor ^ value
```

File: src/operations.rs [hunk 3]
```diff
@@ -123,6 +123,7 @@ pub(crate) fn aesenc(value: u128, xor: u128) -> u128 {
     use core::arch::aarch64::*;
     #[cfg(target_arch = "arm")]
     use core::arch::arm::*;
+    use core::mem::transmute;
```
Comment (joshlf) [inline src/operations.rs:126]: If you use replace the remaining `transmute`s with `transmute!`, you don't need this, and can get rid of more `unsafe` code.

File: src/operations.rs [hunk 4]
```diff
@@ -156,6 +157,7 @@ pub(crate) fn aesdec(value: u128, xor: u128) -> u128 {
...
```
Comment (joshlf) [inline src/operations.rs:160]: If you use replace the remaining `transmute`s with `transmute!`, you don't need this, and can get rid of more `unsafe` code.

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/operations.rs (hunk 3 above)
Comment (PureWhiteWu) [reply src/operations.rs]: Thanks for your review and suggestions!

**Comment by joshlf** [comment]
If you're feeling really ambitious, maybe you could add CI jobs for arm and aarch64? @tkaitchuck would have to say whether they're interested in taking such a change, but I assume they'd be interested.

**Comment by PureWhiteWu** [comment]
@joshlf Hi, I'd love to, but I'm not quite sure if github action provides aarch64 machines?

**Review by tkaitchuck — APPROVED** [review]

File: src/operations.rs [hunk 5]
```diff
@@ -123,10 +123,10 @@ pub(crate) fn aesenc(value: u128, xor: u128) -> u128 {
     use core::arch::aarch64::*;
     #[cfg(target_arch = "arm")]
     use core::arch::arm::*;
-    unsafe {
-        let value = transmute!(value);
-        xor ^ transmute::<_, u128>(vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))))
-    }
+    let value = transmute!(value);
+    let res = unsafe { vaesmcq_u8(vaeseq_u8(value, transmute!(0u128))) };
```
Comment (joshlf) [inline src/operations.rs:127]: Optionally, you could alternatively inline here (I know I suggested the current text lol):

```suggestion
    let res = unsafe { vaesmcq_u8(vaeseq_u8(transmute!(value), transmute!(0u128))) };
```

File: src/operations.rs [hunk 6]
```diff
@@ -156,10 +156,10 @@ pub(crate) fn aesdec(value: u128, xor: u128) -> u128 {
     use core::arch::aarch64::*;
     #[cfg(target_arch = "arm")]
     use core::arch::arm::*;
-    unsafe {
-        let value = transmute!(value);
-        xor ^ transmute::<_, u128>(vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))))
-    }
+    let value = transmute!(value);
+    let res = unsafe { vaesimcq_u8(vaesdq_u8(value, transmute!(0u128))) };
```
Comment (joshlf) [inline src/operations.rs:160]: Optionally:

```suggestion
    let res = unsafe { vaesimcq_u8(vaesdq_u8(transmute!(value), transmute!(0u128))) };
```

**Review by joshlf — APPROVED** [review]

**Pull request merged**